File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:

        sn = nums.sort()
        seen = {} #num, freq

        for x in range(len(nums)):
            seen[nums[x]] = seen.get(nums[x], 0) + 1
        
        topn = k

        sorted_by_values_desc = dict(sorted(seen.items(), key=lambda item: item[1], reverse=True))
        
        return list(sorted_by_values_desc.keys())[:k]




        return


           
        

File: Data Structures & Algorithms/squares-of-a-sorted-array/submission-0.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        

        int l=0;
        int r=nums.length-1;
        
        int[] result = new int[nums.length];

        int last = nums.length-1;

        while (l<=r){

            int sr = nums[r] * nums[r];
            int sl = nums[l] * nums[l];

            if (sr >= sl){
                result[last] = sr;
                r--;
            }
            else{
                result[last] = sl;
                l++;
            }
            last--;
        }

        return result;
    }
}